### DISCUSSION TITLE
Deployer 7 - How do I get the current stage ?

### DISCUSSION THREAD

**Discussion by franck-grenier** [deployphp/deployer#2804] [General]
Hello Deployer,
working on 7 beta, I would like to get the current stage from my launched command :

```bash
dep debug stage=preprod
```

Then, in my tasks, how do I get the asked stage ? I tried several possibilities : 

```php
task('server:task_on_admin', function () {
    info('A specific task on ADMIN servers');
    var_dump(input()->getArgument('stage')); // KO
    var_dump(currentHost()->get('labels')['stage']); // works but not ok, I want the param from the command
    var_dump(get('stage')); // KO
})->select('roles=admin');

```

Thanks for your help !

**Comment by franmomu** [comment]
I think you can use `get('labels')['stage']`

**Reply by antonmedv** [reply]
Yes, this will gave you correct label for specific host.

**Reply by franck-grenier** [reply]
Yes, it works.

But I don't want this one. I'd rather get the argument from the command.

With Deployer 6, we used to call `input()->getArgument('stage')`.

**Reply by antonmedv** [reply]
In v7 it's no longer available. You still can get this argument: `input()->getArgument('selector')` but only as string.

For example, what stage will be in this case `dep deploy all`? Or in this stage `dep deploy stage=prod,stage=beta`?

If you need global param during whole deployment, consider using separate `option('stage', ...);` or env params:

```
STAGE=prod dep deploy stage=$STAGE
```

```php
getenv('STAGE');
```

**Reply by franck-grenier** [reply]
Ok, I will get it from the current host.

My thought was that I could eventually get the wrong stage with erroneous selectors.

But that's maybe some kind of overkill security...

Thanks for the help !
